### ISSUE TITLE
July 27 Update (3.50.0)

### ISSUE THREAD

**Issue by pawREP** [pawREP/ZHM5Randomizer#34]
~~Potentially~~ Confirmed unsupported, needs to be looked at.

**Comment by Diablo990123** [comment]
any eta for the new update?

**Comment by pawREP** [comment]
> any eta for the new update?

Yes, later today

**Comment by Britprogamer** [comment]
> 
> 
> > any eta for the new update?
> 
> Yes, later today

Any update??

**Comment by Britprogamer** [comment]
really want the randomizer back guys. game is just boring now.

**Comment by Diablo990123** [comment]
is there a way to update it  or do we have to wait and hope?

**Comment by Britprogamer** [comment]
Seems like its an issue with the coding... when they did the new update i dont know if some of the code in the engine was changed that the randomizer overrides and as such now doesnt work... or if such code was removed from the game we just dont know..

**Comment by warriorstar-orion** [comment]
Hey pawREP, I have a bunch of changes I'd like to contribute but I don't know if you're made any headway finding memory offsets for the new releases; I'm just continually reverting my installed version of H3 to 3.40.1 to keep using the mod.

Maybe it's worth waiting until all of the Seven Deadly Sins releases are out? Either way, I have some new features:

1. Switching to TOML for the configuration
2. Using spdlog for nice file/console logging when needed
3. A "custom" randomizer that allows players to specify what items they want added to the randomizer right in the config file (e.g. "allow SMGs and any item with the word 'syringe' in them, only allow NPCs to spawn with shotguns")
4. Refactorings to move all the randomizers into separate files to keep things organized

Either way, looking forward to your progress!

Cheers!

**Comment by Diablo990123** [comment]
@warriorstar-orion  how do you revert H3?

**Comment by warriorstar-orion** [comment]
@Diablo990123 i have windows 10's file backup system enabled. Whenever the Epic Store downloads a new update, I just tell Windows 10 to revert the Hitman 3 directory to the version it has backed up.

**Comment by Britprogamer** [comment]
> 
> 
> @Diablo990123 i have windows 10's file backup system enabled. Whenever the Epic Store downloads a new update, I just tell Windows 10 to revert the Hitman 3 directory to the version it has backed up.

wish i had that enabled before 1.50 update

**Comment by warriorstar-orion** [comment]
Hey @pawREP , with the end of the 7 Deadly Sins DLC, would you be interested in trying to find new memory offsets for the various mod hooks? If I can help in any way I'd be happy to.

**Comment by kou4ben3** [comment]
It seems that the developer hasn't been active in a couple months, which is unfortunate because I would have loved to play this.

**Comment by warriorstar-orion** [comment]
I attempted to update the mod to work with the 3.70.0 release of Hitman 3. I've published it here: https://github.com/warriorstar-orion/ZHM5Randomizer/releases/tag/v3.70.0-preview

**Comment by Britprogamer** [comment]
> I attempted to update the mod to work with the 3.70.0 release of Hitman 3. I've published it here: https://github.com/warriorstar-orion/ZHM5Randomizer/releases/tag/v3.70.0-preview

Have tried the new update and it doesnt work for me, says i need to update to latest version, i have 3.70.0.0. game version and your new randomizer.

**Comment by warriorstar-orion** [comment]
@Britprogamer what's the exact error message you get?

**Comment by Britprogamer** [comment]
<img width="1920" alt="Untitled" src="https://user-images.githubusercontent.com/85564207/144760330-088d1cb5-3458-4807-af47-2d6d53185e99.png">

@warriorstar-orion that message

**Comment by warriorstar-orion** [comment]
@Britprogamer  My suspicion is that this is a different version of the game than I get from the Epic Games Store; I'm not sure if there are still separate binaries for DX11 and DX12, but that's my first guess. If that's the case there's not much I can do without being able to install the version of the game you have.

**Comment by Britprogamer** [comment]
latest version off the epic store, using dx 12 binary. Dont know what else to say lol

**Comment by warriorstar-orion** [comment]
Based on that screenshot it looks like you're running Windows 11. It's possible that there were changes from Windows 10 that required a separate version of the binary. Unfortunately, I can't confirm for myself.

**Comment by Britprogamer** [comment]
yes that could be the issue, im not rolling back to 10 though so i'll guess i'll wait a little bit longer.

**Comment by ghost** [comment]
yeah, I get the same issue as Brit listed above. I'm on Win 10 and the latest version of H3

**Comment by warriorstar-orion** [comment]
Are you both on x64 hardware? Or x86?

**Comment by ghost** [comment]
x64 on my end

**Comment by Britprogamer** [comment]
x64 here too

**Comment by warriorstar-orion** [comment]
@BoothTX @Britprogamer  After completely wiping and reinstalling H3, I finally got the same binary as you. Try out https://github.com/warriorstar-orion/ZHM5Randomizer/releases/tag/v3.70.0-preview2.

**Comment by ghost** [comment]
@warriorstar-orion Played about 10 minutes and everything checks out! I appreciate the work you're putting into this :)

**Comment by Britprogamer** [comment]
Works perfectly. Still not working on holiday hoarders map but i think thats because the items the bandits pick up are in the AI script and not the level generation...

**Comment by robertkent87** [comment]
No longer working
https://imgur.com/mY2N0kG

**Comment by warriorstar-orion** [comment]
Yeah, all of the new releases for the Year 2 content are going to break this. I'm waiting until the dust settles for all of that before updating this, so probably once Freelancer drops and is updated for any consequence IOI patches

**Comment by AGENT4787** [comment]
> Yeah, all of the new releases for the Year 2 content are going to break this. I'm waiting until the dust settles for all of that before updating this, so probably once Freelancer drops and is updated for any consequence IOI patches

cmon bro fix the mod pls

**Comment by warriorstar-orion** [comment]
> > Yeah, all of the new releases for the Year 2 content are going to break this. I'm waiting until the dust settles for all of that before updating this, so probably once Freelancer drops and is updated for any consequence IOI patches
> 
> 
> 
> cmon bro fix the mod pls

As I said, I plan to, but not until IOI releases all their Year 2 content. Unlike other mods, every time a new Hitman binary is released, I have to go through the disassembled HITMAN.exe and find the new memory locations of the code we hijack to replace items. This is a time consuming process and I don't want to have to do it for one binary only to have them release another a month later and do it again.

But don't worry, it will happen. I love this mod 🙂

**Comment by quandaledongle** [comment]
> 
Hey year 2 just dropped🤩🤩🤩

**Comment by quandaledongle** [comment]
> > > Yeah, all of the new releases for the Year 2 content are going to break this. I'm waiting until the dust settles for all of that before updating this, so probably once Freelancer drops and is updated for any consequence IOI patches
> > 
> > 
> > cmon bro fix the mod pls
> 
> As I said, I plan to, but not until IOI releases all their Year 2 content. Unlike other mods, every time a new Hitman binary is released, I have to go through the disassembled HITMAN.exe and find the new memory locations of the code we hijack to replace items. This is a time consuming process and I don't want to have to do it for one binary only to have them release another a month later and do it again.
> 
> But don't worry, it will happen. I love this mod 🙂

that will take years when this roadmap is compled year 3 will come and so on there are alot of mod creators on platforms like nexus who keep there mod updated cant this one be updated too?
(i get if you are too busy cuz u ofc also have a life but am just asking love this mod)
